### DISCUSSION TITLE
Running ethernaut in local network

### DISCUSSION THREAD

**Discussion by Prog-block** [OpenZeppelin/ethernaut#578] [Q&A]
Can anyone suggest me a youtube video or any website which can help me to run ethernaut in the local network. Currently I was watching this [video](https://www.youtube.com/watch?v=XXz6zelbjvM) and was unable to run this command `yarn start:ethernaut` at 3 min.

when running this a command the following error was showing:

Error: error:0308010C:digital envelope routines::unsupported
at new Hash (node:internal/crypto/hash:71:19)
at Object.createHash (node:crypto:133:10)
at module.exports (/root/folder/ethernaut/ethernaut/node_modules/webpack/lib/util/createHash.js:135:53)
at NormalModule._initBuildHash (/root/folder/ethernaut/ethernaut/node_modules/webpack/lib/NormalModule.js:417:16)
at handleParseError (/root/folder/ethernaut/ethernaut/node_modules/webpack/lib/NormalModule.js:471:10)
at /root/folder/ethernaut/ethernaut/node_modules/webpack/lib/NormalModule.js:503:5
at /root/folder/ethernaut/ethernaut/node_modules/webpack/lib/NormalModule.js:358:12
at /root/folder/ethernaut/ethernaut/node_modules/loader-runner/lib/LoaderRunner.js:373:3
at iterateNormalLoaders (/root/folder/ethernaut/ethernaut/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
at iterateNormalLoaders (/root/folder/ethernaut/ethernaut/node_modules/loader-runner/lib/LoaderRunner.js:221:10)
/root/folder/ethernaut/ethernaut/node_modules/react-scripts/scripts/start.js:19
throw err;
^

Error: error:0308010C:digital envelope routines::unsupported
at new Hash (node:internal/crypto/hash:71:19)
at Object.createHash (node:crypto:133:10)
at module.exports (/root/folder/ethernaut/ethernaut/node_modules/webpack/lib/util/createHash.js:135:53)
at NormalModule._initBuildHash (/root/folder/ethernaut/ethernaut/node_modules/webpack/lib/NormalModule.js:417:16)
at /root/folder/ethernaut/ethernaut/node_modules/webpack/lib/NormalModule.js:452:10
at /root/folder/ethernaut/ethernaut/node_modules/webpack/lib/NormalModule.js:323:13
at /root/folder/ethernaut/ethernaut/node_modules/loader-runner/lib/LoaderRunner.js:367:11
at /root/folder/ethernaut/ethernaut/node_modules/loader-runner/lib/LoaderRunner.js:233:18
at context.callback (/root/folder/ethernaut/ethernaut/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
at /root/folder/ethernaut/ethernaut/node_modules/react-scripts/node_modules/babel-loader/lib/index.js:59:103 {
opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
library: 'digital envelope routines',
reason: 'unsupported',
code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v18.12.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Can someone please help with this error or suggest a new source.

**Comment by daithi-coombes** [comment]
I'm getting the same error but only difference is I"m running as normal user, as opposed to root (I presume your running as root from stack trace `/root/...`)

node `v18.4.2`
yarn `v1.22.19`
Ubuntu `20.04.6 LTS` (ubuntu studio)

**Comment by daithi-coombes** [comment]
this fixed it for me...
https://stackoverflow.com/a/69699772

tldr, run:
```
NODE_OPTIONS=--openssl-legacy-provider yarn start:ethernaut
```
